AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Operations/Helpers/HostedUISignInHelper.swift

```diff
@@ -26,8 +26,15 @@ struct HostedUISignInHelper: DefaultLogger {
 
     func initiateSignIn() async throws -> AuthSignInResult {
         try await isValidState()
-        log.verbose("Start signIn flow")
-        return try await doSignIn()
+        do {
+            log.verbose("Start signIn flow")
+            let result = try await doSignIn()
+            log.verbose("Received result")
+            return result
+        } catch {
+            await waitForSignInCancel()
```

Comment:
There is a waitForSignIn inside doSignIn function, calling this again might cause necessary logic. 
Maybe consolidate both inside initiateSignIn function?

Reply:
Removed the other `waitForSignInCancel ` and have it at only this place.